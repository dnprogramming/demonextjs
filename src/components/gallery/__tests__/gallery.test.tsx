import { act, render } from "@testing-library/react";
import Gallery from "../gallery";
import React from "react";
import GalleryContent from "@/content/gallery";

describe("Gallery Component", () => {
    it("renders gallery with images", () => {
        const { getByAltText } = render(<Gallery images={GalleryContent} />);
        const imageElement = getByAltText(GalleryContent[0]);
        expect(imageElement).toBeInTheDocument();
    });

    it("renders gallery with click event", () => {
        const { getByAltText } = render(<Gallery images={GalleryContent} />);
        const imageElement = getByAltText(GalleryContent[0]);
        act(() => {
            imageElement.click();
        });
        expect(imageElement).toBeInTheDocument();
    });

    it("opens and closes modal", () => {
        const { getByAltText, getByText } = render(<Gallery images={GalleryContent} />);
        const imageElement = getByAltText(GalleryContent[0]);
        act(() => {
            imageElement.click();
        });
        const closeButton = getByText("Ã—");
        act(() => {
            closeButton.click();
        });
        expect(imageElement).toBeInTheDocument();
    });

    it("renders gallery component unchanged", () => {
        const { container } = render(<Gallery images={GalleryContent}/>);
        expect(container).toMatchSnapshot();
    });
});