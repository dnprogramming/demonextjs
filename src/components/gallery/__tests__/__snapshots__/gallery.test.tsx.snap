// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Gallery Component renders gallery component unchanged 1`] = `
<div>
  <div
    class="galleryGrid"
  >
    <div
      class="imageContainer"
    >
      <img
        alt="/images/hero_banner.png"
        class="thumbnail"
        data-nimg="1"
        decoding="async"
        height="1080"
        loading="lazy"
        src="/_next/image?url=%2Fimages%2Fhero_banner.png&w=3840&q=75"
        srcset="/_next/image?url=%2Fimages%2Fhero_banner.png&w=1920&q=75 1x, /_next/image?url=%2Fimages%2Fhero_banner.png&w=3840&q=75 2x"
        style="color: transparent; cursor: pointer;"
        width="1920"
      />
    </div>
  </div>
</div>
`;
